expr/interval_year_to_month.go

```diff
@@ -0,0 +1,36 @@
+package expr
+
+import (
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+type intervalYearMonthVal struct {
```

Comment:
Same feedback as other literal. Let's just create a Literal struct impl that is public and give it utility methods as appropriate.

Reply:
fixed it